src/pygeom.c

```diff
@@ -216,12 +216,12 @@ PyTypeObject GeometryType = {
 /* Get a GEOSGeometry pointer from a GeometryObject, or NULL if the input is
 Py_None. Returns 0 on error, 1 on success. */
 char get_geom(GeometryObject *obj, GEOSGeometry **out) {
-    if (!PyObject_IsInstance((PyObject *) obj, (PyObject *) &GeometryType)) {
+    PyTypeObject *type = ((PyObject *)obj)->ob_type;
+    if ((type != &GeometryType) & (type->tp_base != &GeometryType)) {
         if ((PyObject *) obj == Py_None) {
             *out = NULL;
             return 1;
         } else {
-            PyErr_Format(PyExc_TypeError, "One of the arguments is of incorrect type. Please provide only Geometry objects.");
```

Comment:
Other alternative would be also to re-acquire the gil here, I suppose

Reply:
(athough reading below, I suppose the templating makes this a bit more unified)

Reply:
The python macro actually includes a brace `}` so it seems to be intended to have the GIL release and re-acquisition in the same function.